sonic_cli_gen/generator.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+import os
+import pkgutil
+import jinja2
+
+from sonic_cli_gen.yang_parser import YangParser
+
+templates_path_switch = '/usr/share/sonic/templates/sonic-cli-gen/'
+
+
+class CliGenerator:
+    """ SONiC CLI generator. This class provides public API
+    for sonic-cli-gen python library. It can generate config,
+    show CLI plugins.
+
+        Attributes:
+            logger: logger
+    """
+
+    def __init__(self, logger):
+        """ Initialize CliGenerator. """
+
+        self.logger = logger
+
+
+    def generate_cli_plugin(
+        self,
+        cli_group,
+        plugin_name,
+        config_db_path='configDB',
+        templates_path='/usr/share/sonic/templates/sonic-cli-gen/'
+    ):
+        """ Generate click CLI plugin and put it to:
+            /usr/local/lib/<python>/dist-packages/<CLI group>/plugins/auto/
+        """
+
+        parser = YangParser(
+            yang_model_name=plugin_name,
+            config_db_path=config_db_path,
+            allow_tbl_without_yang=True,
+            debug=False
+        )
+        # yang_dict will be used as an input for templates located in
+        # /usr/share/sonic/templates/sonic-cli-gen/
+        yang_dict = parser.parse_yang_model()
+
+        loader = jinja2.FileSystemLoader(templates_path)
+        j2_env = jinja2.Environment(loader=loader)
+        template = j2_env.get_template(cli_group + '.py.j2')
+
+        plugin_path = get_cli_plugin_path(cli_group, plugin_name + '_yang.py')
+
+        with open(plugin_path, 'w') as plugin_py:
```

Comment:
We allow overwriting, if exist. Is that OK?

Reply:
Yes, because it is not OK to have 2 CLI modules with different names but the same content, because it will lead to warning messages and incorrect flows when executing `config` or `show` CLI commands.